Documentation/ExtensionArchitecture/FileStructure/ComposerJson.rst

```diff
@@ -201,82 +156,73 @@ require
 
 (*required*)
 
-At the least, you will want to require `typo3/cms-core`.
-You should add other system extensions and third party extensions,
-if your extension depends on them.
+At least, you will need to require `typo3/cms-core` in the according version(s).
+You should add other system extensions and third-party extensions, if your
+extension depends on them.
 
 In Composer-based installations the loading order of extensions and their
-dependencies is derived from :js:`require` and :js:`suggest`
+dependencies is derived from `require` and `suggest`.
 
 
 
 suggest
 -------
 
-You should add other system extensions and third party extensions,
-if your extension has an optional dependency on them.
+You should add other system extensions and third-party extensions, if your
+extension has an optional dependency on them.
 
 In Composer-based installations the loading order of extensions and their
-dependencies is derived from :js:`require` and :js:`suggest`
+dependencies is derived from `require` and `suggest`.
 
 
 autoload
 --------
 
 (*required*)
 
-Define namespace - path mapping for PSR-4 autoloading.
-In TYPO3 we follow the convention that all classes (except test classes)
-are in the directory :file:`Classes`.
+The autoload section defines the namespace/path mapping for
+`PSR-4 autoloading <https://www.php-fig.org/psr/psr-4/>`. In TYPO3 we follow
+the convention that all classes (except test classes) are in the directory
+:file:`Classes/`.
 
-extra
------
+extra.typo3/cms.extension-key
+-----------------------------
 
 (*required*)
 
 Not providing this property will emit a deprecation notice and will fail in
 future versions.
 
-.. hint::
-
-   The property "extension-key" means the **literal string** "extension-key",
-   not your actual extension key. The value on the right side should be your
-   actual extension key.
+..  hint::
+    The property `extension-key` means the **literal string** `extension-key`,
+    not your actual extension key. The value on the right side should be your
+    actual extension key.
 
-Example for extension key **bootstrap_package**:
+Example for extension key `my_extension`:
 
-.. code-block:: json
+..  literalinclude:: _ComposerJson/_ExtensionKey.json
+    :language: json
+    :caption: Excerpt of EXT:my_extension/composer.json
 
-   {
-      "extra": {
-         "typo3/cms": {
-            "extension-key": "bootstrap_package"
-         }
-      }
-   }
 
 Properties no longer used
 =========================
 
 version
 -------
 
-Was used in earlier TYPO3 versions.
-For TYPO3 versions 7.6 and above you should not use the version property.
-The version for the extension is set in the file :ref:`ext_emconf.php <ext_emconf-php>`.
+This was used in earlier TYPO3 versions. For TYPO3 versions 7.6 and above you
+should not use the version property. The version for the extension is set in
+the file :ref:`ext_emconf.php <ext_emconf-php>`.
```

Comment:
In composer-based installations the ext_emconf.php is not mandatory anymore since TYPO3 v11, so this sentence rings wrong

Reply:
I suggest to remove the version section altogether. No one working with Packagist should use version ever in a composer.json. TYPO3 version 7.6 is also rather old.

Reply:
yes, lets remove it!

Reply:
Done